src/NUnitFramework/tests/TestDataAssemblyTests.cs

```diff
@@ -0,0 +1,38 @@
+// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System.Diagnostics;
+using System.Reflection;
+
+namespace NUnit.Framework
+{
+    public static class TestDataAssemblyTests
+    {
+        [Test(Description = "Needed because we test stack traces and don’t want to have to put [MethodImpl(MethodImplOptions.NoInlining)] on everything involved.")]
+        public static void AssemblyIsNotOptimized()
+        {
+            Assert.That(Assembly.Load(new AssemblyName("nunit.testdata")),
+                Has.Attribute<DebuggableAttribute>().With.Property("IsJITOptimizerDisabled").EqualTo(true));
```

Comment:
Good idea.

Reply:
Charlie's. =)

Reply:
Cool! Most likely no great overhead since the assembly will already be in memory.